app/controllers/houses_controller.rb

```diff
@@ -3,11 +3,14 @@ class HousesController < ApplicationController
 
   # GET /houses or /houses.json
   def index
-    @houses = House.all
+    @houses = current_user.houses
   end
 
   # GET /houses/1 or /houses/1.json
   def show
+    unless HousePolicy.new(current_user, @house).show?
+      raise Pundit::NotAuthorizedError, "not allowed"
+    end
```

Comment:
You should be able to replace this with 

```suggestion
    authorize @house
```

[since you included Pundit in application controller](https://chapters.firstdraft.com/chapters/860#convention-over-configuration-in-controllers).